### PR TITLE
Made sure that multiple repos can be updated at once,

### PR DESCRIPTION
**Pull request by saleem-latif**

### CODE REVIEW

File: src/main/java/org/intellij/sdk/codesync/HandleBuffer.java [hunk 1]
```diff
@@ -158,50 +168,51 @@ public static void handleBuffer() {
...
-                        System.out.printf("Failed to connect to websocket endpoint: %s.\n", WEBSOCKET_ENDPOINT);
-                    }
-                } else {
+        // Send Diffs in a single request.
+        ConfigRepo configRepo = configFile.getRepo(currentRepo);
+        CodeSyncWebSocketClient codeSyncWebSocketClient = client.getWebSocketClient(configRepo.token);
+        codeSyncWebSocketClient.connect(isConnected -> {
+            if (isConnected) {
+                try {
+                    codeSyncWebSocketClient.sendDiffs(diffsToSend, (successfullyTransferred, diffFilePath) -> {
```
Comment (Basit-qc) [inline src/main/java/org/intellij/sdk/codesync/HandleBuffer.java:201]: 👍

File: src/main/java/org/intellij/sdk/codesync/clients/CodeSyncWebSocketClient.java [hunk 2]
```diff
@@ -93,23 +105,79 @@ public void sendDiff(DiffFile diffFile, Integer fileId, DataTransmissionHandler
...
+            diff.put("is_binary", diffFile.isBinary);
+            diff.put("created_at", Utils.formatDate(diffFile.createdAt));
+            diff.put("path", diffFile.fileRelativePath);
+            diff.put("diff_file_path", diffFile.originalDiffFile.getPath());
+
+            diffs.add(diff);
+        }
+
+        JSONObject payload = new JSONObject();
+        payload.put("diffs", diffs);
```
Comment (Basit-qc) [inline src/main/java/org/intellij/sdk/codesync/clients/CodeSyncWebSocketClient.java:158]: Are we sending chunk of diffs in one send call? (i.e. more than 1 diff per sendDiff request?)

Comment (saleem-latif) [reply src/main/java/org/intellij/sdk/codesync/clients/CodeSyncWebSocketClient.java:158]: yes

Comment (Basit-qc) [reply src/main/java/org/intellij/sdk/codesync/clients/CodeSyncWebSocketClient.java:158]: Ok, in JS & brew daemons we are sending 1 diff per send request to handle it independently. However server design is flexible, can accept multiple diffs. But we need to see there is no duplication. e.g. 
1- How many diffs are sent in 1 request?
2- Say if 8/10 are successfuly processed by server but 2/10 encountered an error, what will happen with those 2 diffs?

Comment (Basit-qc) [reply src/main/java/org/intellij/sdk/codesync/clients/CodeSyncWebSocketClient.java:158]: Ah, I see
 ```String diffFilePath = (String) response.get("diff_file_path");``` which means failed diffs will remain in buffer and will be tried again in next iteration. Did I get this right?

Comment (Basit-qc) [reply src/main/java/org/intellij/sdk/codesync/clients/CodeSyncWebSocketClient.java:158]: If this is working well, we probably can use same optimization in JS sides as well where callback is async.

Comment (saleem-latif) [reply src/main/java/org/intellij/sdk/codesync/clients/CodeSyncWebSocketClient.java:158]: Yes, you are correct. I did not see any errors in my testing so I think we are good to implement it in the JS side as well.

**Pull request merged**